pkg/actuators/machine/conditions.go

```diff
@@ -15,3 +15,64 @@ limitations under the License.
 */
 
 package machine
+
+import (
+	ibmcloudproviderv1 "github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klog "k8s.io/klog/v2"
+)
+
+const (
+	machineCreationSucceedReasonCondition  = "MachineCreationSucceeded"
+	machineCreationSucceedMessageCondition = "Machine successfully created"
+	machineCreationFailedReasonCondition   = "MachineCreationFailed"
+)
+
+// reconcileProviderConditions updates condition for the machine and returns new Condition []
+func reconcileProviderConditions(conditions []ibmcloudproviderv1.IBMCloudMachineProviderCondition, newCondition ibmcloudproviderv1.IBMCloudMachineProviderCondition) []ibmcloudproviderv1.IBMCloudMachineProviderCondition {
+	currTime := metav1.Now()
+
+	currCondition := conditionTypeCheck(conditions, newCondition.Type)
+
+	if currCondition == nil {
+		klog.Infof("Adding new provider condition %v", newCondition)
+		conditions = append(conditions,
+			ibmcloudproviderv1.IBMCloudMachineProviderCondition{
+				Type:               newCondition.Type,
+				Status:             newCondition.Status,
+				Reason:             newCondition.Reason,
+				Message:            newCondition.Message,
+				LastTransitionTime: currTime,
+				LastProbeTime:      currTime,
+			},
+		)
+	} else {
+		// Update if new Status is diff from existing Status
+		// Update if new Message is diff from existing Message
+		// Update if new Reason is diff from existing Reason
+		if currCondition.Status != newCondition.Status || currCondition.Message != newCondition.Message || currCondition.Reason != newCondition.Reason {
+			klog.Infof("Updating provider condition %v", newCondition)
+			currCondition.Status = newCondition.Status
+			currCondition.Reason = newCondition.Reason
+			currCondition.Message = newCondition.Message
+			currCondition.LastProbeTime = currTime
+
+			// Update LastTransitionTime if Status differ
+			if currCondition.Status != newCondition.Status {
```

Comment:
The `Status` delta check here between current vs new to set the transition time should occur prior to the assignment of the current status to new on line 56